### PR TITLE
Feature/contactme

### LINKED ISSUE #3: As an user, I want to Contact the owner of the page through a contact form
- Create a form for name, email and message and a submit button.
- Make sure that required fields are filled with proper syntax.
- Check if I've received a message.

### PR DESCRIPTION
**Pull request by aikkuvo333**

### CODE REVIEW

**Comment by elenabli** [comment]
First of all I read the README file and  I want to note that it is very clearly written.
The structure of the project is logical, following the rules of writing React projects.
Second, not related to this pullrequest: I noticed that the .env file is visible in the remote repository, this should not happen. You need to get that file into gitignore. 
image

To the pull request:
There are several commits logically separated from each other. Comments of commits are not overloaded with text, but clearly reflect the essence. 
The names of variables, functions and components are also well thought out, the code is easy to read and to understand.

Improvements: you have multiple components and each has its own css file. This is a good practice, in my humble opinion. But it would be better to separate components with their css into folders.

Nice work!

**Comment by aikkuvo333** [comment]
Thanks for the feedback! I will hide my env file and sort the files into smaller sections as suggested

File: src/App.css [hunk 1]
```diff
@@ -22,13 +22,4 @@
     background-repeat: no-repeat;
     color: #fff;
     font-size: 100px;
-  }
-  
-  .contact {
-    background-image: url('/Users/aivanvo/DX project/react-website-portfolio/public/images/image2.jpg');
```
Comment (pennane) [inline src/App.css:28]: There are still files with absolute paths based on your own computer. This means that the project cannot be opened anywhere other than on your device.

Consider changing the urls to relative ones.

e.g
```css
background-image: url('../public/images/image2.jpg');
```

File: src/components/ContactForm.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    email: '',
+    message: '',
+  });
+
+  const handleChange = (e) => {
+    const { name, value } = e.target;
+    setFormData({
+      ...formData,
+      [name]: value,
+    });
```
Comment (pennane) [inline src/components/ContactForm.js]: when modifying react state based on the old state it is often beneficial to provide a function for the setter.

This way the old version of the state is more likely to behave as expected

```suggestion
    setFormData((old) => ({
      ...old,
      [name]: value,
    }));
```

File: src/pages/Contact.css [hunk 3]
```diff
@@ -0,0 +1,8 @@
+.contact {
+    background-image: url('/Users/aivanvo/DX project/react-website-portfolio/public/images/image2.jpg');
```
Comment (pennane) [inline src/pages/Contact.css:2]:
```suggestion
    background-image: url('../../public/images/image2.jpg');
```

Comment (pennane) [reply src/pages/Contact.css:2]: ( this might have some implications when the project is bundled with webpack etc )

File: src/pages/Contact.js [hunk 4]
```diff
@@ -1,10 +1,35 @@
...
-    <h1 className='contact'>
-      Contact me from here 
-    </h1>
+    <div>
+      <div>
+        <h1>Contact</h1>
+        <p>Want to work together on a project?</p>
+      </div>
+      <div>
+        <ContactForm></ContactForm>
```
Comment (pennane) [inline src/pages/Contact.js:14]: this is more of a preference thing, but if an element is provided no children, then the component can be used in a self closing manner
```suggestion
        <ContactForm />
```

**Review by pennane — COMMENTED** [review]
Seems like this is still the last weeks pull request. Nevertheless, I added some code level comments.

**Review by stenvma81 — COMMENTED** [review]
I haven't looked at this repo yet so I will begin with general comments...

- The structure is nice and straight-forward. As the amount of components increases, it might be a good idea to split them into sub folders to keep the project tidy. Same goes for the pages.
- The .gitignore should be planned carefully at the start of the project. A service like Toptals .gitignore generator can be used for a good template but it always pays off to do some research and make it yourself.
- As far as the naming, coding conventions etc. goes the app looks quite nice.
- Some of names in .css files could be a bit more descriptive. This is nitpicking, but consider if it makes sense to shorten for ex button to btn. It's more of a personal preference too I guess, but still something to consider. 

**About the PR**

- The user story (I presume it's the one from last week) is not properly linked to the PR. You can link it in the Development cell on the right tab of the PR. The story itself is nice and explains the feature being worked on. Consider adding a need that is being fulfilled by addressing the story: As a user I want to contact the owner of the page to discuss future collaborations.
- As the user story is not linked, the PR itself should contain some info on what was worked on and why.

@pennane already brought up some code level nice issues.

Good work!

**Comment by aikkuvo333** [comment]
Thanks for taking the time and giving such thorough feedbacks guys, i appreciate it. I agree a lot of the general and code improvements that were mentioned and will apply them in the future.

**Pull request closed without merging**